pants-plugins/schemas/rules.py

```diff
@@ -49,37 +42,24 @@ class GenerateSchemasFieldSet(FieldSet):
 
 class GenerateSchemasViaFmtTargetsRequest(FmtTargetsRequest):
     field_set_type = GenerateSchemasFieldSet
-    name = CMD
+    tool_subsystem = GenerateSchemas
+    partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
 
 
 @rule(
     desc="Update contrib/schemas/*.json with st2-generate-schemas",
     level=LogLevel.DEBUG,
 )
 async def generate_schemas_via_fmt(
-    request: GenerateSchemasViaFmtTargetsRequest,
+    request: GenerateSchemasViaFmtTargetsRequest.Batch,
+    subsystem: GenerateSchemas,
 ) -> FmtResult:
     # We use a pex to actually generate the schemas with an external script.
     # Generation cannot be inlined here because it needs to import the st2 code.
```

Comment:
Does this comment make sense now we have removed this chunk of code, as the external script isnt' directly referenced here anymore?

Reply:
This code block was moved, not removed. The PexFromTargetsRequest (including all of the random constants used to build it) are now defined on the subsystem. I added a comment to help clarify that.